snippets/components/video-render.js

```diff
@@ -0,0 +1,147 @@
+        <style>
+          video {
+            max-width: 100%;
+            aspect-ratio: ${aspectRatio};
+          }
+        </style>
+        
+        <video muted autoplay></video>  
+      `;
+
+    if (!this.shadowRoot) {
+      this.attachShadow({ mode: "open" });
+      this.shadowRoot.append(this.videoElement.content.cloneNode(true));
+    }
+
+    this._isLoaded = false;
+    this.observer = null;
+    this.update();
+  }
+
+  connectedCallback() {
+    this.setObserver();
+  }
+
+  disconnectedCallback() {
+    this.observer?.disconnect();
+  }
+
+  get isLoaded() {
+    return this._isLoaded;
+  }
+
+  set isLoaded(value) {
+    this._isLoaded = value;
+  }
+
+  setObserver() {
+    this.observer = new IntersectionObserver((entries) => {
+      const video = this.shadowRoot.querySelector("video");
+
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          this.handleVideoInView(video);
+        } else {
+          this.handleVideoOutOfView(video);
+        }
+      });
+    });
+
+    this.observer.observe(this);
+  }
+
+  handleVideoInView(video) {
+    if (!this.isLoaded) {
+      this.loadVideo(video);
+      this.observer.unobserve(video);
+    } else {
+      video.play();
+    }
+  }
+
+  handleVideoOutOfView(video) {
+    if (this.isLoaded) {
+      video.pause();
+    }
+  }
+
+  loadVideo(video) {
+    Array.from(video.children).forEach((child) => {
+      if (child.tagName === "source".toUpperCase()) {
+        child.src = child.getAttribute("data-src");
+      }
+    });
+
+    video.load();
+    this.isLoaded = true;
+  }
+
+  update() {
+    const videoTag = this.shadowRoot.querySelector("video");
+    const videoSources = JSON.parse(this.getAttribute("sources"));
+
+    videoSources.forEach((source) => {
+      let sourceTag = document.createElement("source");
+
+      sourceTag.setAttribute("data-src", source.url);
+      sourceTag.setAttribute("type", source.mime_type);
+
+      videoTag.append(sourceTag);
+    });
+
+    this.addVideoAttributes(videoTag);
+    this.removeComponentAttributes();
+
+    if (videoTag.getAttribute("fullwidth")) this.setFullWidth(videoTag);
+  }
+
+  addVideoAttributes(videoTag) {
+    const allAtributes = this.getAttributeNames();
+
+    allAtributes.forEach((attribute) => {
+      const value = this.getAttribute(attribute);
+      videoTag.setAttribute(attribute, value);
+    });
+
+    this.removeVideoAttributes();
+  }
+
+  removeVideoAttributes() {
+    const attributes = ["class", "sources", "width", "height"];
+
+    attributes.forEach((attribute) => {
+      this.shadowRoot.querySelector("video").removeAttribute(attribute);
+    });
+  }
+
+  removeComponentAttributes() {
+    const attributes = ["width", "height", "part", "fullwidth", "loop"];
+
+    attributes.forEach((attribute) => {
+      this.removeAttribute(attribute);
+    });
+  }
+
+  getAspectRatio(width, height) {
+    if (width === "auto" || height === "auto") return "auto";
```

Comment:
```suggestion
    if (width == null || height == null) return "auto";
    if (width === "auto" || height === "auto") return "auto";
```

A minor addition, always good to check for errors. Using `== null` will check for `null` and `undefined` and can avoid potential issue if either were not to be set properly. This is a "just in case" kind of check.